controllers/pkg/reconcilers/generic-specializer/reconciler.go

```diff
@@ -286,13 +288,24 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		kptfile := rl.Items.GetRootKptfile()
 		if kptfile == nil {
 			r.recorder.Event(pr, corev1.EventTypeWarning, "ReconcileError", "mandatory Kptfile is missing")
-			r.l.Error(fmt.Errorf("mandatory Kptfile is missing from the package"), "")
+			r.l.Error(err, "mandatory Kptfile is missing from the package")
 			return ctrl.Result{}, nil
 		}
 
+		r.l.Info("generic specializer root kptfile", "packageName", pr.Spec.PackageName, "repository", pr.Spec.RepositoryName, "kptfile", kptfile)
+
 		kptf := kptfilelibv1.KptFile{Kptfile: rl.Items.GetRootKptfile()}
 		pr.Status.Conditions = porchcondition.GetPorchConditions(kptf.GetConditions())
+		/*
```

Comment:
If we don't need to update the PR at all, we can clean out some of this code. Fine perhaps for now to just put a TODO in with an explanation.